packages/lab/src/slider/internal/useKeyDownThumb.ts

```diff
@@ -0,0 +1,57 @@
+import type { SliderChangeHandler, SliderValue } from "../types";
+import { clampValue, roundToStep, setValue } from "./utils";
+
+const getValueFromKeyName = (
+  keyName: string,
+  value: number,
+  min: number,
+  max: number,
+  step: number,
+) => {
+  switch (keyName) {
+    case "Home":
+      return min;
+    case "End":
+      return max;
+    case "ArrowUp":
+    case "ArrowRight":
+      return value + step;
+    case "ArrowDown":
+    case "ArrowLeft":
+      return value - step;
+    default:
+      return value;
+  }
+};
+
+export const useKeyDownThumb = (
+  min: number,
+  max: number,
+  step: number,
+  value: SliderValue,
+  onChange: SliderChangeHandler,
+  index: number,
+) => {
+  return (event: React.KeyboardEvent) => {
```

Comment:
is prevent default needed for page scrolling?

Reply:
Good catch. Added the `preventDefault` and also changed it to only accept the key events we need (`Home`, `End`, `ArrowUp`, `ArrowRight`, `ArrowDown`, `ArrowLeft`)